app/src/main/java/com/example/taskyy/ui/AgendaScreen.kt

```diff
@@ -1,12 +1,203 @@
 package com.example.taskyy.ui
 
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.AbsoluteRoundedCornerShape
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDefaults
+import androidx.compose.material3.DisplayMode
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
+import androidx.compose.material3.rememberDatePickerState
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.example.taskyy.R
+import com.example.taskyy.domain.navigation.Screen
+import com.example.taskyy.ui.events.AgendaEvent
+import com.example.taskyy.ui.viewmodels.AgendaState
+import java.time.LocalDateTime
+import java.time.ZoneId
 
 @Composable
-fun AgendaScreen(){
-    Column {
-        Text(text = "Agenda Screen")
+fun AgendaScreen(state: AgendaState, navController: NavController, onEvent: (AgendaEvent) -> Unit){
```

Comment:
A NavController reference can also quickly break your Compose preview which is why we usually keep that outside the screen composables. Since you already provide an `onEvent` lambda here (which is good!), you can listen to these events in the `composable` block of your NavHost and navigate there:
```
composable("agenda) {
    AgendaScreen(
        state = viewModel.state,
        onEvent = { event ->
            when(event) {
                is AgendaEvent.Logout -> // navigate to login
                is AgendaEvent.OnAgendaItemClick -> // navigate to agenda details
                ...
            }
            viewModel.onEvent(event) // Still send all events to the VM so nothing is lost
        }
    )
}
```